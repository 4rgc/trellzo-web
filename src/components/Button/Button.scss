%btn-base {
	text-align: center;
	border: none;
	padding: 5px;
	letter-spacing: 1.3px;
	border-radius: 2px 7px 2px 7px;
	height: 36px;
	text-transform: uppercase;
	font-weight: 300;
	word-spacing: 70%;
	cursor: pointer;
}

%btn-dis {
	cursor: default;
}

%btn-prm {
	color: #fbfffe;
	background-color: seagreen;
}

%btn-sec {
	color: rgb(64, 182, 115);
	background-color: rgba(251, 255, 254, 0.2);
	font-weight: 600;
}

%btn-prm-dis {
	@extend %btn-dis;
	background-color: rgba(64, 182, 115, 0.2);
	color: grey;
}

%btn-sec-dis {
	background-color: rgba(251, 255, 254, 0.2);
	color: rgba(64, 182, 115, 0.4);
	font-weight: 600;
}

%btn-sq {
	min-width: 40px;
}

%btn-sm {
	min-width: 70px;
}

%btn-md {
	min-width: 140px;
}

%btn-lg {
	min-width: 200px;
}

%btn-reg {
	height: 36px;
}

%btn-thn {
	height: 30px;
}

$sizes: ('sq', 'sm', 'md', 'lg');
$kinds: ('prm', 'prm-dis', 'sec', 'sec-dis');
$heights: ('reg', 'thn');

@each $size in $sizes {
	@each $kind in $kinds {
		@each $height in $heights {
			.btn-#{$kind}-#{$size}-#{$height} {
				@extend %btn-base;
				@extend %btn-#{$size};
				@extend %btn-#{$kind};
				@extend %btn-#{$height};
			}
		}
	}
}
