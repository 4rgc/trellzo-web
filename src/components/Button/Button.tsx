import './Button.scss';

export interface IButtonProps extends React.ComponentPropsWithoutRef<'button'> {
	size?: 'square' | 'small' | 'medium' | 'large';
	kind?: 'primary' | 'secondary';
	height?: 'regular' | 'thin';
	disabled?: boolean;
	onClick?: () => void;
	children?: React.ReactChild | React.ReactChild[];
}

const Button: React.FC<IButtonProps> = (props) => {
	const {
		disabled = false,
		children,
		onClick,
		kind = 'primary',
		size = 'medium',
		height = 'regular',
		className,
	} = props;

	const kindToClass = { primary: 'prm', secondary: 'sec' };
	const disabledToClass = '-dis';
	const sizeToClass = {
		square: 'sq',
		small: 'sm',
		medium: 'md',
		large: 'lg',
	};
	const heightToClass = { regular: 'reg', thin: 'thn' };

	const buttonClass = `btn-${kindToClass[kind]}${
		disabled ? disabledToClass : ''
	}-${sizeToClass[size]}-${heightToClass[height]} ${className || ''}`;

	return (
		<button
			className={buttonClass}
			onClick={disabled ? undefined : onClick}
		>
			{children}
		</button>
	);
};

export default Button;
