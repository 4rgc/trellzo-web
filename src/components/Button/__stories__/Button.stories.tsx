import { ComponentStory, ComponentMeta } from '@storybook/react';

import Button from '../Button';

export default {
	title: 'Components/Button',
	component: Button,
	argTypes: { onClick: { action: 'clicked' } },
} as ComponentMeta<typeof Button>;

const Template: ComponentStory<typeof Button> = ({ children, ...args }) => (
	<Button {...args}>{children}</Button>
);

export const Default = Template.bind({});

export const Square = Template.bind({});
Square.args = { size: 'square', children: 'OK' };

export const Small = Template.bind({});
Small.args = { size: 'small', children: 'Small' };

export const Medium = Template.bind({});
Medium.args = { size: 'medium', children: 'Medium Btn' };

export const Large = Template.bind({});
Large.args = { size: 'large', children: 'Large Button !!!' };

export const Primary = Template.bind({});
Primary.args = { kind: 'primary', children: 'Primary' };

export const Secondary = Template.bind({});
Secondary.args = { kind: 'secondary', children: 'Secondary' };

export const Disabled = Template.bind({});
Disabled.args = { disabled: true, children: 'disabled' };

export const Thin = Template.bind({});
Thin.args = { height: 'thin' };

export const CustomClass = Template.bind({});
CustomClass.args = {
	disabled: false,
	children: 'fake disabled',
	className: 'btn-dis',
};
