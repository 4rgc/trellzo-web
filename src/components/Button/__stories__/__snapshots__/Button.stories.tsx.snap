// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Button CustomClass smoke-test 1`] = `
<button class="btn-prm-md-reg btn-dis">
  fake disabled
</button>
`;

exports[`Components/Button Default smoke-test 1`] = `
<button class="btn-prm-md-reg ">
</button>
`;

exports[`Components/Button Disabled smoke-test 1`] = `
<button class="btn-prm-dis-md-reg ">
  disabled
</button>
`;

exports[`Components/Button Large smoke-test 1`] = `
<button class="btn-prm-lg-reg ">
  Large Button !!!
</button>
`;

exports[`Components/Button Medium smoke-test 1`] = `
<button class="btn-prm-md-reg ">
  Medium Btn
</button>
`;

exports[`Components/Button Primary smoke-test 1`] = `
<button class="btn-prm-md-reg ">
  Primary
</button>
`;

exports[`Components/Button Secondary smoke-test 1`] = `
<button class="btn-sec-md-reg ">
  Secondary
</button>
`;

exports[`Components/Button Small smoke-test 1`] = `
<button class="btn-prm-sm-reg ">
  Small
</button>
`;

exports[`Components/Button Square smoke-test 1`] = `
<button class="btn-prm-sq-reg ">
  OK
</button>
`;

exports[`Components/Button Thin smoke-test 1`] = `
<button class="btn-prm-md-thn ">
</button>
`;
